yaksh/models.py

```diff
@@ -1077,11 +1083,16 @@ def regrade(self, question_id):
         json_data = question.consolidate_answer_data(answer) \
                             if question.type == 'code' else None
         correct, result = self.validate_answer(answer, question, json_data)
-        user_answer.marks = question.points if correct else 0.0
         user_answer.correct = correct
         user_answer.error = result.get('error')
+        if correct:
```

Comment:
If we have a fraction(value 0 to 1) returned from the code sever, then we do not need  this if block, we can simply write 
```user_answer.marks = question.points * result['marks']```
if wrong then marks is 0
if right then marks is 1
else fraction of marks.
So no need to check correct or wrong for partial grading when adding marks.